server/src/main/java/com/epam/aidial/core/server/log/GfLogStore.java

```diff
@@ -189,10 +192,22 @@ private void append(ProxyContext context, LogEntry entry, String assembledStream
     }
 
     private static void append(LogEntry entry, Buffer buffer) {
-        if (buffer != null) {
-            byte[] bytes = buffer.getBytes();
-            String chars = new String(bytes, StandardCharsets.UTF_8); // not efficient, but ok for now
-            append(entry, chars, true);
+        if (buffer == null) {
+            return;
+        }
+        boolean largeBuffer;
+        if (exceedLimit(buffer)) {
+            buffer = buffer.slice(0, MAX_BODY_SIZE_BYTES);
```

Comment:
You can break utf-8 encoded tail. One character can be encoded as 4 bytes in utf-8.

Reply:
String constructor uses decoder that fixes this broken tail. Could you write a test to see what will be in this case?

Reply:
good catch!

Reply:
The decoder will replace with character (65533). So it won't drop an exception at least.

Reply:
The decoder replaces broken bytes with 65533 character. 

It won't throw an exception.